protocol/nats_jetstream/v2/receiver.go

```diff
@@ -89,10 +90,20 @@ func NewConsumerFromConn(conn *nats.Conn, stream, subject string, jsmOpts []nats
 
 	streamInfo, err := jsm.StreamInfo(stream, jsmOpts...)
 
+	subjectMatch := stream + ".*"
+	if !strings.HasPrefix(subject, stream) {
+		// Use an empty subject parameter in conjunction with
+		// nats.ConsumerFilterSubjects
+		subjectMatch = ""
+	} else if strings.Count(subject[len(stream):], ".") > 1 {
+		// More than one "." in the remainder of subject, use ".>" to match
+		subjectMatch = stream + ".>"
+	}
```

Comment:
Question (minor): is there a case where both conditions could be true? If not, this code could be simplified by removing the `else` and just having two `if` statements to increase readability without if/else indentation?

Reply:
Also, once #1083 lands (depending on the implementation by @stephen-totty-hpe), we need to make sure to also apply this logic to the new JetStream implementation.

Reply:
This code could also be written as:

```golang
if strings.HasPrefix(subject, stream) {
  if strings.Count(strings.TrimPrefix(subject, stream), ".") > 1 {
    // Change .* to .>
  }
} else {
  // Change to ""
}
```

I think if stream is `foo` and "subject" is `bar.baz.12345`, then the reformulated code would hit the second condition.

We _could_ rewrite this as:

```golang
if strings.HasPrefix(subject, stream) && strings.Count(strings.TrimPrefix(subject, stream), ".") > 1 {
  // Change .* to .>
}
if !strings.HasPrefix(subject, stream) {
  subjectMatch = ""
}

But, we need the `HasPrefix` check on the `strings.Count`.  I could switch to using `TrimPrefix` (or some other tricks, like getting the prefix-trimmed string, and comparing its length to the original to determine `HasPrefix`), but none of the above seemed better (in particular, `TrimPrefix` in the non-prefixed case returns the original string, which seemed wrong.

Thinking further, I suppose I could remove the `HasPrefix` from the first clause, as the second will completely overwrite if `HasPrefix` is false.